plugins/techdocs/src/reader/transformers/styles/rules/layout.ts

```diff
@@ -42,7 +42,7 @@ export default ({ theme, sidebar }: RuleOptions) => `
 .md-nav__icon {
   height: 20px !important;
   width: 20px !important;
-  margin-left:${theme.spacing(1)}px;
+  margin-left: ${theme.spacing(1)};
```

Comment:
This doesn't quite work right?

Reply:
Hm why not? Note that this is in one large interpolated string.

Reply:
It would break for cases where `theme.spacing(1)` returns `8`, since this is CSS the unit is required

Reply:
Oh right and since it's in a string literal rather than having jss as an intermediary ... I see. So this one has no real fix,  besides some hack checking if its number-like (doesn't have a unit) and adding if missing? Hmmm. OK I can add that, if it's what's needed

Reply:
Done

Reply:
Ah, having dug into this I realized what the issue is 😅, I was a bit confused since this really shouldn't be an issue, v4 consumers should get a v4 theme with v4 spacing, and same for v5. We need to not mutate the theme here: https://github.com/backstage/backstage/blob/ae5fa463d7b6585a0c0e4748e0cfc64a83382610/packages/theme/src/unified/overrides.ts#L87, and instead just override the theme spacing locally for the `style` call.